File: Behavioural-Patterns/Mediator/C#/Mediator/Implementations/NotificationService.cs
public class NotificationService : Subscriber {

    public void onEvent(Object ev) {
        if (ev is UserLoggedInEvent) {
            UserLoggedInEvent loginEvent = (UserLoggedInEvent) ev;
           Console.WriteLine("   [NotificationService] Sent push notification: Welcome back, " + loginEvent.username + "!");
        }
    }

    public String getName() {
        return "NotificationService";
    }
    
}

File: Behavioural-Patterns/Mediator/Java/Implementations/NotificationService.java
package Implementations;

import Interfaces.Subscriber;

public class NotificationService implements Subscriber {

    @Override
    public void onEvent(Object event) {
        if (event instanceof UserLoggedInEvent) {
            UserLoggedInEvent loginEvent = (UserLoggedInEvent) event;
            System.out.println("   [NotificationService] Sent push notification: Welcome back, " + loginEvent.username + "!");
        }
    }

    @Override
    public String getName() {
        return "NotificationService";
    }
    
}
